src/hazelcore/semantics/UHPat.re

```diff
@@ -153,6 +158,51 @@ and make_inconsistent_operand =
     (Parenthesized(set_p), u_gen);
   };
 
+let patterns_of_type =
+    (u_gen: MetaVarGen.t, ty: HTyp.t): (list(t), MetaVarGen.t) =>
+  OpSeq.(
+    switch (ty) {
+    | HTyp.Hole
+    | Unit
+    | Arrow(_, _) => ([], u_gen)
```

Comment:
in these cases maybe generate a pattern containing just a single hole, instead of the empty list?

Reply:
Well, in these cases there are just no patterns to generate. I thought returning an empty list would be clearer about this than one with just a single hole pattern.

Reply:
you can technically pattern match on arrow, its just that the only valid patterns are wildcard and variable. but leaving a hole for this is fine. not a big deal but a naive implementation of the next step would have to special case this situation to avoid removing all the rules in the case.

Reply:
Okay, I'll keep it in mind.